crates/sui-types/src/temporary_store.rs

```diff
@@ -1148,7 +1128,7 @@ impl<S: GetModule + ObjectStore + BackingPackageStore> TemporaryStore<S> {
             // not in input objects, must be a dynamic field
             let obj = self
                 .store
-                .get_object(id)
+                .get_object_by_key(id, expected_version)
```

Comment:
panic / expect if None?

Reply:
I believe that is what the code is and was doing right?
`map_err` changes to an `invariant_violation`, it is `?` and the resulting `Option` is `ok_or_else` which gives the `invariant_violation` on `None`.
At least that is my understanding

Reply:
also version protected the code so that version mismatch is kept for backward compatibility